### PR TITLE
Set to Kryo serializer when registerKryoSerialization is called

### PR DESCRIPTION
**Pull request by nwangtw**
In Storm, users just need to register class serializer. Kryo is the default serializer. However in Heron, the default is Java serializer which doesn't have the functionality. I think it is more consistent to set to KryoSerializer when the registration function is called.

Another option is to set Kryo serializer as the default one. The concern is that some users might be affected by the implicit change.

I am totally open to discussion.

### CODE REVIEW

File: heron/api/src/java/org/apache/heron/api/Config.java [hunk 1]
```diff
@@ -480,11 +480,19 @@ private static List getRegisteredKryoSerializations(Map<String, Object> conf) {
...
+    // Set to Kryo Seralizer when this function is caled. In Storm API, Kryo seralizer is
+    // the default but it is not in Heron. This implicit set could make migration easier
+    // since it is more consistent.
+    setSerializationClassName("org.apache.heron.api.serializer.KryoSerializer");
     getRegisteredKryoSerializations(conf).add(klass.getName());
   }
 
   public static void registerKryoSerialization(
       Map<String, Object> conf, Class klass, Class<? extends Serializer> serializerClass) {
+    // Set to Kryo Seralizer when this function is caled. In Storm API, Kryo seralizer is
```
Comment (huijunwu) [inline heron/api/src/java/org/apache/heron/api/Config.java]: `called`

File: heron/api/src/java/org/apache/heron/api/Config.java [hunk 2]
```diff
@@ -480,11 +480,19 @@ private static List getRegisteredKryoSerializations(Map<String, Object> conf) {
   }
 
   public static void registerKryoSerialization(Map<String, Object> conf, Class klass) {
+    // Set to Kryo Seralizer when this function is caled. In Storm API, Kryo seralizer is
```
Comment (huijunwu) [inline heron/api/src/java/org/apache/heron/api/Config.java]: `called`

**Comment by huijunwu** [comment]
The Heron API is not bound to Storm API. The translation is done by Heron-Storm API. Ideally, the Storm compatibility is supposed to be done in Heron-Storm API.

File: heron/api/src/java/org/apache/heron/api/Config.java [hunk 3]
```diff
@@ -480,11 +480,19 @@ private static List getRegisteredKryoSerializations(Map<String, Object> conf) {
...
+    // the default but it is not in Heron. This implicit set could make migration easier
+    // since it is more consistent.
+    setSerializationClassName("org.apache.heron.api.serializer.KryoSerializer");
```
Comment (nlu90) [inline heron/api/src/java/org/apache/heron/api/Config.java]: create a final field for this `"org.apache.heron.api.serializer.KryoSerializer"` string and reuse it everywhere

**Comment by nwangtw** [comment]
> The Heron API is not bound to Storm API. The translation is done by Heron-Storm API. Ideally, the Storm compatibility is supposed to be done in Heron-Storm API.

Yeah. The translation works (although hacky). This is for native Heron API which works a little differently from the Heron-Storm API.

File: heron/api/src/java/org/apache/heron/api/Config.java (hunk 3 above)
Comment (nwangtw) [reply heron/api/src/java/org/apache/heron/api/Config.java]: kk. done.

**Pull request merged**